src/geojson/FeatureCollection.cpp

```diff
@@ -7,15 +7,75 @@ Feature FeatureCollection::get_feature(int index) const {
     return features[index];
 }
 
+int FeatureCollection::find(Feature feature) {
+    for(int feature_index = 0; feature_index < this->get_size(); feature_index++) {
+        if (*this->features[feature_index] == *feature) {
+            return feature_index;
+        }
+    }
+
+    return -1;
+}
+
+int FeatureCollection::find(std::string ID) {
```

Comment:
This would be a good place to keep an `unorded_map<std::string, int> index_by_id` and then just look up the vector index by ID.
If the memory overhead of the map is significant, a refresh of the overall storage mechanics might be in order.